python/lsst/meas/extensions/shapeHSM/_hsm_moments.py

```diff
@@ -130,21 +140,25 @@ def _calculate(
         """
         # Convert centroid to GalSim's PositionD type.
         guessCentroid = galsim.PositionD(centroid.x, centroid.y)
-
         try:
             # Attempt to compute HSM moments.
-            shape = galsim.hsm.FindAdaptiveMom(
+
+            # Use galsim c++/python interface directly.
+            shape = galsim.hsm.ShapeData()
+            hsmparams = galsim.hsm.HSMParams.default
+
+            galsim._galsim.FindAdaptiveMomView(
```

Comment:
Even this isn't really guaranteed to be stable, but it should be very close to equivalent to `image.FindAdaptiveMom(...)`.

Reply:
So there was actually a significant amount of time spent in `_convertMask` which was some of the motivation to use the primitive.  If there was an officially supported way of bypassing this I think we could use the official API.

Reply:
Oh right. The `_convertMask` [function ](https://github.com/GalSim-developers/GalSim/blob/releases/2.5/galsim/hsm.py#L483-L529)has a bunch of `galsim.Image` instantiations which makes it slow as well. I could open an issue on the GalSim repo if there's no immediate supported API.

Reply:
We supply a value for `badpix` that masks some pixels, so very much real. I think the overheads there can be reduced by using `galsim._ImageI` instead of `galsim.ImageI` in `_convertMask`.